### ISSUE TITLE
Multiple button support?

### ISSUE THREAD

**Issue by HackShitUp** [huri000/SwiftEntryKit#19]
Is there a way to display multiple buttons on top of each other in such a way that they're stacked in the view? I briefly saw the documentation and saw that action-methods are supported, but how would one pass n-number of buttons, each with different delegate methods?

**Comment by huri000** [comment]
Hi,
You can use `EKProperty.ButtonBarContent` as data source of `EKButtonBarView`.
`EKButtonBarView` receives `EKProperty.ButtonBarContent` instance and inits itself accordingly.
If there are more than 2 buttons - they automatically stacked vertically as in the alert presets in the demo app.

`EKProperty.ButtonBarContent` instance contains an array of `EKProperty.ButtonContent` - one for each button. Each has an action.
More than one action (.touchUpInside) is currently not supported. But i'm open to suggestions if that's not enough IYO :)

**Comment by HackShitUp** [comment]
Could this only be achieved with EKAlertMessageView?

**Comment by huri000** [comment]
That's an interesting point.
`EKAlertMessageView` only provides UI infrastructure for using it. It uses `expandAnimately` to expand the buttons by changing inner priorities.
Since `EKButtonBarView` doesn't handle the expansion itself, you need to do it yourself by doing:
```Swift
buttonBarView.expand()
buttonBarView.layoutIfNeeded()
```
I should move the `expand` logic to the button bar view itself sometime soon. I have no doubt that logic should be there and not in its container.

**Comment by HackShitUp** [comment]
Right, I was using the UI debugger on Xcode and setting the EKAttributes from 

``` swift
attributes.positionConstraints.size = .init(width: .offset(value: 20), height: .intrinsic)
```

to 

``` swift
attributes.positionConstraints.size = .init(width: .offset(value: 20), height: EKAttributes.PositionConstraints.Edge.constant(value: 100))
```

displays the buttons, but not vertically stacked with the third button. Also, using 

``` swift
buttonBarView.expand()
```

logs this error: **'expand' is inaccessible due to 'internal' protection level**

**Comment by HackShitUp** [comment]
By the way, I think this is the _best_ presentation libraries out there on this planet 👍

**Comment by huri000** [comment]
Thanks a lot! ^^
I'll review the button bar view anyway and expose more of its methods.

**Comment by huri000** [comment]
I revised EKButtonBarView's `expand` method to contain all the expansion related logic.
Now, you basically only need to create an `EKProperty.ButtonBarContent` instance, init`EKButtonBarView` using it. Then call display method on `SwiftEntryKit` using your attributes.
It's available on version 0.3.3.

**Comment by huri000** [comment]
@HackShitUp 
B.T.W: You still need to invoke `expand` on the `EKButtonBarView` instance.
I believe the requirements specified in this issue are satisfied. Let me know if you think otherwise. 
Thanks!

**Comment by Smiller193** [comment]
@huri000 can I see an example of this please

**Comment by huri000** [comment]
@Smiller193 - The example project contains everything you need to know. :-)

**Comment by Smiller193** [comment]
I just want a center popup example with multiple buttons

**Comment by Smiller193** [comment]
@huri000 I see the example with one button but not two

**Comment by huri000** [comment]
Have you tried the Service Rating preset?

**Comment by Smiller193** [comment]
umm let me try

**Comment by HackShitUp** [comment]
@huri000 I think if you could provide an example in the README, it could really help out

**Comment by huri000** [comment]
That’s a good idea 💡
I’m aware that some of the abilities of the libraries are not clear enough
to users. But its Readme has just became too large to be added with
additional information. I plan to build a Wiki for the presets and their
any additional stuff.

**Comment by SumitSMARTER** [comment]
@huri000  I am new to Swift and was trying to use your Library, what I was trying to achieve is the same thing which @HackShitUp, multiple buttons in the EKPopUpmessage.
I checked the Rating example but, what I perceive that it has **buttonBarContent: <#T##EKProperty.ButtonBarContent#>)**, but the PopUp msg does not have that property, EKPopUpMessage(themeImage: <#T##EKPopUpMessage.ThemeImage?#>, 
title: <#T##EKProperty.LabelContent#>, 
description: <#T##EKProperty.LabelContent#>, 
**button: <#T##EKProperty.ButtonContent#>**, 
action<#T##EKPopUpMessage.EKPopUpMessageAction##EKPopUpMessage.EKPopUpMessageAction##() -> ()#>)

Please help how to accomplish this, I am ClueLess, if there is any example, it would be much Helpful.

Thanks.

**Comment by swarajsinghdev** [comment]
still not any solution
